src/Utils.hh

```diff
@@ -204,6 +204,58 @@ namespace sdf
     return errors;
   }
 
+  /// \brief Load all objects of a specific sdf element type. No error
+  /// is returned if an element is not present.
+  /// \param[in] _sdf The SDF element that contains zero or more elements.
+  /// \param[in] _sdfName Name of the sdf element, such as "model".
+  /// \param[in] _config Specific parser configuration to use to load
+  /// \param[out] _objs Elements that match _sdfName in _sdf are added to this
+  /// vector, unless an error is encountered during load.
+  /// \return The vector of errors. An empty vector indicates no errors were
+  /// experienced.
+  template <typename Class>
+  sdf::Errors loadRepeated(sdf::ElementPtr _sdf, const std::string &_sdfName,
+      const ParserConfig &_config,
+      std::vector<Class> &_objs,
+      const std::function<void(Class &)> &_beforeLoadFunc = {})
```

Comment:
I looked and I can't find a single place where this `std::function` argument is used. `git blame` indicates that it was added in #381, which was a big PR, and I'm guessing we didn't notice that the argument wasn't used by the time it was ready to be merged.

Given that `_beforeLoadFunc` isn't used, I would suggest removing the parameter and the small code block where it is used, and adding `const ParserConfig &_config = ParserConfig::GlobalConfig()` in its place so we don't need to duplicate this method.

Reply:
ok, but now I see that the `Load(obj, _config)` call is different, so we need two different versions of this template function. never mind I guess

Reply:
Yeah, I took a moment to try and do some SFINAE template trickery to dispatch to the correct signature, but it ended up getting ugly quickly for such a simple function and I opted to just duplicate.

If all of the DOM objects eventually have a Load(obj, config), then we could remove the original method, but that would increase the scope of this pr substantially (for not much benefit at the moment)

Reply:
👍

Reply:
hold on, I believe [loadUniqueRepeated](https://github.com/gazebosim/sdformat/blob/sdf13/src/Utils.hh#L104-L105) is able to handle this with `Args&&... _args`, since [World::Load passes a ParserConfig to Model::Load](https://github.com/gazebosim/sdformat/blob/sdf13/src/World.cc#L225)

that would require removing the `beforeLoadFunc` argument, but it's not used anyway

Reply:
Good call, I switched to the variadic implementation in [1746843](https://github.com/gazebosim/sdformat/pull/1147/commits/17468435d3b288f01eb20d22e27842b9b15c46f8)